### PR TITLE
Document FieldInfo.SetValue breaking change

### LINKED ISSUE #3851: FieldBuilder.SetValue for static init-only fields in .NET Core 3.0+
The documentation says:

https://github.com/dotnet/dotnet-api-docs/blob/eb7cd6ec7fe65807ae94c55bc152a128a8224014/xml/System.Reflection.Emit/FieldBuilder.xml#L941

For static init-only fields, this workaround was working in .NET Framework, however this is no longer supported in .NET Core 3.0+ (see https://github.com/dotnet/runtime/issues/11571#issuecomment-442675193).

So, how to initialize static init-only field of dynamic type in .NET Core 3.0+?
It would be great if the documentation will provide this hint:)

### PR DESCRIPTION
**Pull request by jkotas**
## Summary

Document breaking change https://github.com/dotnet/runtime/issues/11571

Fixes #3851

### CODE REVIEW

File: xml/System.Reflection/FieldInfo.xml [hunk 1]
```diff
@@ -2096,6 +2096,8 @@
 ## Remarks  
  This method will assign `value` to the field reflected by this instance on object `obj`. If the field is static, `obj` will be ignored. For non-static fields, `obj` should be an instance of a class that inherits or declares the field. The new value is passed as an `Object`. For example, if the field's type is Boolean, an instance of `Object` with the appropriate Boolean value is passed. Before setting the value, `SetValue` checks to see if the user has access permission. This final method is a convenience method for calling the following `SetValue` method.  
   
+ This method cannot be used to set value of static init-only (`readonly` in C#) fields reliably. In .NET Core 3.0+, attempt to set value of static init-only field causes exception to be thrown.
```
Comment (gewarren) [inline xml/System.Reflection/FieldInfo.xml]:
```suggestion
This method cannot be used to set values of static init-only (`readonly` in C#) fields reliably. In .NET Core 3.0 and later versions, an exception is thrown if you attempt to set a value on a static init-only field.
```

File: xml/System.Reflection/FieldInfo.xml [hunk 2]
```diff
@@ -2195,6 +2197,8 @@
   
 ## Remarks  
  This method will assign `value` to the field reflected by this instance on `obj`. If the field is static, `obj` will be ignored. For non-static fields, `obj` should be an instance of a class that inherits or declares the field. The new value is passed as an `Object`. For example, if the field's type is `Boolean`, an instance of `Object` with the appropriate Boolean value is passed. Before setting the value, `SetValue` checks to see if the user has access permission.  
+
+ This method cannot be used to set value of static init-only (`readonly` in C#) fields reliably. In .NET Core 3.0+, attempt to set value of static init-only field causes exception to be thrown.
```
Comment (gewarren) [inline xml/System.Reflection/FieldInfo.xml]:
```suggestion
This method cannot be used to set values of static init-only (`readonly` in C#) fields reliably. In .NET Core 3.0 and later versions, an exception is thrown if you attempt to set a value on a static init-only field.
```

File: xml/System.Reflection/FieldInfo.xml [hunk 3]
```diff
@@ -2096,6 +2096,8 @@
 ## Remarks  
  This method will assign `value` to the field reflected by this instance on object `obj`. If the field is static, `obj` will be ignored. For non-static fields, `obj` should be an instance of a class that inherits or declares the field. The new value is passed as an `Object`. For example, if the field's type is Boolean, an instance of `Object` with the appropriate Boolean value is passed. Before setting the value, `SetValue` checks to see if the user has access permission. This final method is a convenience method for calling the following `SetValue` method.  
   
+This method cannot be used to set values of static init-only (`readonly` in C#) fields reliably. In .NET Core 3.0 and later versions, an exception is thrown if you attempt to set a value on a static init-only field.
```
Comment (gewarren) [inline xml/System.Reflection/FieldInfo.xml]:
```suggestion
This method cannot be used to set values of static, `init-only` (`readonly` in C#) fields reliably. In .NET Core 3.0 and later versions, an exception is thrown if you attempt to set a value on a static, `init-only` field.
```

File: xml/System.Reflection/FieldInfo.xml [hunk 4]
```diff
@@ -2195,6 +2197,8 @@
   
 ## Remarks  
  This method will assign `value` to the field reflected by this instance on `obj`. If the field is static, `obj` will be ignored. For non-static fields, `obj` should be an instance of a class that inherits or declares the field. The new value is passed as an `Object`. For example, if the field's type is `Boolean`, an instance of `Object` with the appropriate Boolean value is passed. Before setting the value, `SetValue` checks to see if the user has access permission.  
+
+This method cannot be used to set values of static init-only (`readonly` in C#) fields reliably. In .NET Core 3.0 and later versions, an exception is thrown if you attempt to set a value on a static init-only field.
```
Comment (gewarren) [inline xml/System.Reflection/FieldInfo.xml]:
```suggestion
This method cannot be used to set values of static, `init-only` (`readonly` in C#) fields reliably. In .NET Core 3.0 and later versions, an exception is thrown if you attempt to set a value on a static, `init-only` field.
```

File: xml/System.Reflection/FieldInfo.xml (hunk 3 above)
Comment (jkotas) [reply xml/System.Reflection/FieldInfo.xml]: static and init-only are similar kind of qualifiers. I would either put back-ticks around both or neither.

I do not have a strong opinion about it. I would be happy to apply the suggestion if you prefer to have only init-only in back-ticks. Let me know.

Comment (gewarren) [reply xml/System.Reflection/FieldInfo.xml]: In the ECMA spec, they refer to the **initonly** attribute, but when not referring to a keyword, they use init-only as the general term. So I agree, I don't think we need the backticks here. @BillWagner?

https://www.ecma-international.org/publications/files/ECMA-ST/ECMA-335.pdf

Comment (gewarren) [reply xml/System.Reflection/FieldInfo.xml]: But we do need a comma between static and init-only.

Comment (jkotas) [inline xml/System.Reflection/FieldInfo.xml]:
```suggestion
This method cannot be used to set values of static init-only (`readonly` in C#) fields reliably. In .NET Core 3.0 and later versions, an exception is thrown if you attempt to set a value on a static, init-only field.
```

File: xml/System.Reflection/FieldInfo.xml (hunk 4 above)
Comment (jkotas) [inline xml/System.Reflection/FieldInfo.xml]:
```suggestion
This method cannot be used to set values of static, init-only (`readonly` in C#) fields reliably. In .NET Core 3.0 and later versions, an exception is thrown if you attempt to set a value on a static init-only field.
```

File: xml/System.Reflection/FieldInfo.xml [hunk 5]
```diff
@@ -2096,6 +2096,8 @@
 ## Remarks  
  This method will assign `value` to the field reflected by this instance on object `obj`. If the field is static, `obj` will be ignored. For non-static fields, `obj` should be an instance of a class that inherits or declares the field. The new value is passed as an `Object`. For example, if the field's type is Boolean, an instance of `Object` with the appropriate Boolean value is passed. Before setting the value, `SetValue` checks to see if the user has access permission. This final method is a convenience method for calling the following `SetValue` method.  
   
+This method cannot be used to set values of static init-only (`readonly` in C#) fields reliably. In .NET Core 3.0 and later versions, an exception is thrown if you attempt to set a value on a static, init-only field.
```
Comment (jkotas) [inline xml/System.Reflection/FieldInfo.xml]:
```suggestion
This method cannot be used to set values of static, init-only (`readonly` in C#) fields reliably. In .NET Core 3.0 and later versions, an exception is thrown if you attempt to set a value on a static, init-only field.
```

File: xml/System.Reflection/FieldInfo.xml [hunk 6]
```diff
@@ -2195,6 +2197,8 @@
   
 ## Remarks  
  This method will assign `value` to the field reflected by this instance on `obj`. If the field is static, `obj` will be ignored. For non-static fields, `obj` should be an instance of a class that inherits or declares the field. The new value is passed as an `Object`. For example, if the field's type is `Boolean`, an instance of `Object` with the appropriate Boolean value is passed. Before setting the value, `SetValue` checks to see if the user has access permission.  
+
+This method cannot be used to set values of static, init-only (`readonly` in C#) fields reliably. In .NET Core 3.0 and later versions, an exception is thrown if you attempt to set a value on a static init-only field.
```
Comment (jkotas) [inline xml/System.Reflection/FieldInfo.xml]:
```suggestion
This method cannot be used to set values of static, init-only (`readonly` in C#) fields reliably. In .NET Core 3.0 and later versions, an exception is thrown if you attempt to set a value on a static, init-only field.
```

File: xml/System.Reflection/FieldInfo.xml (hunk 3 above)
Comment (jkotas) [reply xml/System.Reflection/FieldInfo.xml]: > But we do need a comma between static and init-only.

Fixed

**Pull request merged**